tests/test_downloader.py

```
@@ -3731,6 +3843,217 @@ def test_error_handling_comprehensive_ui_paths(tmp_path, caplog):
+        # Mock repository to return a newer prerelease
+        mock_fetch_dirs.return_value = ["firmware-1.1.0.fedcba"]
+        mock_fetch_contents.return_value = [
+            {
+                'name': 'firmware.bin',
+                'download_url': 'https://example.com/firmware.bin',
+                'size': 1024
+            }
+        ]
+
+        # Mock successful download
+        def mock_download_func(url, dest):
```

Ruff (0.13.1)

Unused function argument: `url`

(ARG001)